### PR TITLE
allow using VAAPI with OpenGLES

### PR DESCRIPTION
**Pull request by lrusak**
Now with the vpupdates merged I wanted to get this rolling.

The changes are pretty trivial.

This is needed to be able to use VAAPI with OpenGLES (drm/kms, mir, wayland, etc)

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h [hunk 1]
```diff
@@ -22,9 +22,16 @@
 
 #include "system.h"
 
+#ifdef HAS_GL
 #include "cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h"
+using CRendererVAAPIBase = CLinuxRendererGL;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h]: this is too hacky and needs a better solution

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShader.cpp [hunk 2]
```diff
@@ -239,6 +239,10 @@ BaseYUV2RGBGLSLShader::BaseYUV2RGBGLSLShader(bool rect, unsigned flags, EShaderF
     m_defines += "#define XBMC_YV12\n";
   else if (m_format == SHADER_NV12)
     m_defines += "#define XBMC_NV12\n";
+  else if (m_format == RENDER_FMT_CVBREF)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShader.cpp]: of course not, RENDER_FMT_* is history here

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h]: better write your own renderer for gles, having a this one dependent on gl and gles makes no sense

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShader.cpp (hunk 2 above)
Comment (lrusak) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShader.cpp]: fixed

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h (hunk 1 above)
Comment (lrusak) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h]: so would you prefer to have `RendererVAAPIGL.h`, `RendererVAAPIGL.cpp` and `RendererVAAPIGLES.h`, `RendererVAAPIGLES.cpp` ???

Comment (lrusak) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h]: or convert `CLinuxRendererGL` and `CLinuxRendererGLES` to just be `CLinuxRenderer` and decide at build time which one would be used.

**Comment by lrusak** [comment]
@FernetMenta updated.

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPI.h]: that is much better than forcing constraints on base classes. assume a change in LinuxRendererGL. your code here would foce a change in LinuxRendererGLES that in turn forces changes on all of its subclasses. that is not maintainable.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp [hunk 3]
```diff
@@ -2635,12 +2635,14 @@ bool COutput::GLInit()
   }
 #endif
 
+#ifdef HAS_GL
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: why is this required?

Comment (lrusak) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]:
```
/home/lukas/libreelec/build.LibreELEC-Generic.x86_64-9.0-devel/kodi-95a82b7/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp: In member function 'bool VAAPI::COutput::GLInit()':
/home/lukas/libreelec/build.LibreELEC-Generic.x86_64-9.0-devel/kodi-95a82b7/xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp:2641:23: error: 'GL_TEXTURE_RECTANGLE_ARB' was not declared in this scope
     m_textureTarget = GL_TEXTURE_RECTANGLE_ARB;
                       ^~~~~~~~~~~~~~~~~~~~~~~~
```

Comment (lrusak) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: This is about all I can find, http://stackoverflow.com/questions/6883160/opengl-es-gl-texture-rectangle

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: ok, could you please put a comment in the code with reference to this. when I get time I'll check if GL_TEXTURE_RECTANGLE_ARB is still required for GL. I think that can be dropped but not completely sure.

Comment (lrusak) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: added

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
Squashed. Good to go now.

**Comment by FernetMenta** [comment]
+1
